### PR TITLE
feat: improve auth functionalities

### PR DESCRIPTION
**Pull request by devodii**
- add email package with resend helper
- support for credential authentication
- implement password hashing with bcrypt

### CODE REVIEW

File: apps/coursition/src/app/api/auth/[...nextauth]/auth-options.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import type { NextAuthOptions } from 'next-auth'
+import type { Adapter } from 'next-auth/adapters'
+import CredentialsProvider from 'next-auth/providers/credentials'
+import GoogleProvider from 'next-auth/providers/google'
+
+const handleAuthorize = (dto: { email: string; password: string }) =>
+  Effect.gen(function* () {
+    const result = yield* Effect.tryPromise({
+      try: async () => await prisma.user.findFirstOrThrow({ where: { email: dto.email } }),
+      catch: () => Effect.fail('user not found'),
```
Comment (NaucMeIT) [inline apps/coursition/src/app/api/auth/[...nextauth]/auth-options.ts:14]: **blocker:** We don't handle invalid password and failing effect like this is not correct way. We should use Expected Error (mentioned in review above).

File: apps/coursition/src/components/googleSignIn.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      onClick={async () => {
+        setIsLoading(true)
+        await signIn('google', { callbackUrl: '/' })
+      }}
+      className='w-ful text-white font-medium py-2 px-4 rounded-md'
+    >
+      <span>Sign in with google</span>
+      {isLoading && <Loader className='animate-spin text-sm ml-1' />}
+    </Button>
+  )
```
Comment (NaucMeIT) [inline apps/coursition/src/components/googleSignIn.tsx:20]: **nitpick:** We should use server login (that works even without JS. But we'll handle it with Auth.js v5.
https://authjs.dev/getting-started/session-management/login

```suggestion
  return (
    <form
      action={async () => {
        "use server"
        await signIn("google", { callbackUrl: "/" })
      }}
    >
      <button type="submit">Sign in</button>
    </form>
  )
```

File: apps/legacy_nmit/pnpm-lock.yaml
Comment (NaucMeIT) [inline apps/legacy_nmit/pnpm-lock.yaml:1]: **blocker:** You shouldn't delete anything in `legacy_nmit`, it's old version of NMIT and is not part of the ticket.

**Review by NaucMeIT — CHANGES_REQUESTED** [review]
**blocker:** We shouldn't add EffectTS as part of this ticket. It's not well architected and would need another major rewrite.

File: libs/email/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Effect } from 'effect'
```
Comment (BleedingDev) [inline libs/email/src/index.ts:1]: Why is there still Effect? Please remove it completely, including from `package.json`.

File: package.json [hunk 4]
```diff
@@ -19,10 +32,11 @@
     "@radix-ui/react-checkbox": "^1.1.1",
     "@radix-ui/react-label": "^2.1.0",
     "@radix-ui/react-slot": "^1.1.0",
-    "@vercel/remote-nx": "^2.0.0",
     "ai": "^3.3.19",
+    "bcryptjs": "^2.4.3",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
+    "effect": "^3.6.5",
```
Comment (BleedingDev) [inline package.json:39]: Please remove Effect.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

**Pull request merged**